Zend/zend_inheritance.c

```diff
@@ -1565,21 +1565,34 @@ static void zend_do_traits_property_binding(zend_class_entry *ce) /* {{{ */
 					zend_hash_del(&ce->properties_info, prop_name);
 					flags |= ZEND_ACC_CHANGED;
 				} else {
+					not_compatible = 1;
+					
 					if ((coliding_prop->flags & (ZEND_ACC_PPP_MASK | ZEND_ACC_STATIC))
 						== (flags & (ZEND_ACC_PPP_MASK | ZEND_ACC_STATIC))) {
-						/* flags are identical, now the value needs to be checked */
+						/* the flags are identical, thus, the properties may be compatible */
+						zval op1, op2;
+
 						if (flags & ZEND_ACC_STATIC) {
-							not_compatible = fast_is_not_identical_function(&ce->default_static_members_table[coliding_prop->offset],
-											  &ce->traits[i]->default_static_members_table[property_info->offset]);
+							ZVAL_COPY(&op1, &ce->default_static_members_table[coliding_prop->offset]);
```

Comment:
This is not thread-safe ... for 7.2 I'd go with just pointers + ZVAL_DEREF, as we don't actually need the copies here.

Reply:
Going that route, would it be necessary to check any case other than the existing static property? As far as I can tell, it's the only case where it would be a ref.

Reply:
I've pushed a fix based on the assumption above and in line with 7.2